processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -2115,6 +2152,10 @@ private boolean doesNotNeedPresenceCheckForSourceParameter(PropertyMapping mappi
         return mapping.getAssignment().isSourceReferenceParameter();
     }
 
+    public boolean shouldDirectlyReturnOnlyMappingResult() {
+        return this.returnTypeToConstruct.isEnumType() && this.propertyMappings.size() == 1;
```

Comment:
Maybe we could come up with more tests. I have no idea what scenarios could be tested, but after checking the coverage report we could provide a test where `this.propertyMappings.size()` is `!= 1`

Not quite sure though, my coverage report seems to be a bit buggy.